openfeature/client.go

```diff
@@ -659,6 +659,35 @@ func (c *Client) Object(ctx context.Context, flag string, defaultValue interface
 	return value
 }
 
+// Track performs an action for tracking for occurrence  of a particular action or application state.
+//
+// Parameters:
+// - ctx is the standard go context struct used to manage requests (e.g. timeouts)
+// - trackingEventName is the event name to track
+// - evalCtx is the evaluation context used in a flag evaluation (not to be confused with ctx)
+// - trackingEventDetails defines optional data pertinent to a particular
+func (c *Client) Track(ctx context.Context, trackingEventName string, evalCtx EvaluationContext, details TrackingEventDetails) {
```

Comment:
I'm 99% that this is correct, and we want a void function here, but we _could_ also return an error.

I think that returning an error would be wrong since it would violate this spec point: https://openfeature.dev/specification/sections/flag-evaluation/#requirement-1410, but I want to point it out.

Reply:
I also believe that it is better not to return error here, because the client have not do anything that is needed to return error.

In case of some errors such as context cancelled or network-related, the provider should have mechanism to handle them.

Reply:
I agree. A tracking event should be fire and forget.